packages/react-instantsearch/src/core/createConnector.js

```diff
@@ -139,7 +139,7 @@ export default function createConnector(connectorDesc) {
         metadata,
       } = store.getState();
       const searchState = {results, searching, error};
-      return connectorDesc.getProps(props, widgets, searchState, metadata);
+      return connectorDesc.getProps.call(this, props, widgets, searchState, metadata);
```

Comment:
The `this` used in this code should refer to the Element itself. This is the one we want here because we're using an arrow function to bind this method with the instance and it is safe to use it because this is where the state lives as well. So it's like storing data in the `this` of a component.